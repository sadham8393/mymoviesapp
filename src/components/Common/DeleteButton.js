import React from 'react';
import PropTypes from 'prop-types';

const DeleteButton = (props) =>{
    return (
        <button className = "waves-effect waves-light btn-small deleteBtn" onClick = {() => props.onDeleteClick(props.movie)}> Delete </button>
    )
}

DeleteButton.propTypes = { 
    movie: PropTypes.object, 
    onDeleteClick: PropTypes.func
} 

export default DeleteButton;