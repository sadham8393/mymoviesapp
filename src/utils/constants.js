// URl constants

export const MOVIEDB_PATH = "https://api.themoviedb.org/3/";
export const REACT_MOVIEDB_API_KEY  = "?api_key=0e89680c8b3fa0f259b53bbcc8f6091d";
export const MOVIEDB_IMAGE_APPEND_URL = "https://image.tmdb.org/t/p/w500/";
export const TRENDING_DAY_URL = "trending/all/day";


export const NO_IMAGE_URl = "https://s3-ap-southeast-1.amazonaws.com/upcode/static/default-image.jpg";

//Common Constants
export const INPUT_TYPE_TEXT = "text";
export const INPUT_TYPE_SELECT = "select";
export const NOT_AVAILABLE = "Not Available";


//Action Constants
export const FETCH_TRENDING_MOVIES = "FETCH_TRENDING_MOVIES";
export const FETCH_TV_MOVIES = "FETCH_TV_MOVIES";
export const DELETE_TV_MOVIES = "DELETE_TV_MOVIES";
export const MOVIES_FETCHED_SUCCESS = "MOVIES_FETCHED_SUCCESS";
export const MOVIES_FETCHED_ERROR = "MOVIES_FETCHED_ERROR";
export const DELETE_MOVIE_SUCCESS = "DELETE_MOVIE_SUCCESS";
export const DELETE_MOVIE_FAILURE = "DELETE_MOVIE_FAILURE";

export const MOVIES_FETCHED_NO_RESULTS = "MOVIES_FETCHED_NO_RESULTS";

export const RESET_RESPONSE = "RESET_RESPONSE";

export const NO_RESULTS_FOUND = "No results found for the given search input";

export const SERVER_UNAVAILABE = "Server unavailable. Please try after sometime"



