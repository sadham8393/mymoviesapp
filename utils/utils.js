import configureStore from 'redux-mock-store'

export const findByTestClass = (component, className) => {
    const wrapper = component.find(`${className}`);
    return wrapper;
};

export const findByComponent = (component, innerComponent) => {
    const wrapper = component.find(innerComponent);
    return wrapper;
};

export const testStore = (initialState) => {
    let store;
    const mockStore = configureStore();
    store = mockStore(initialState)
    return store;
};