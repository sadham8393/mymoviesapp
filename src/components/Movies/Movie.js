import React from 'react';
import { MOVIEDB_IMAGE_APPEND_URL, NO_IMAGE_URl} from '../../utils/constants';
import DeleteButton from '../Common/DeleteButton';
import PropTypes from 'prop-types';

const Movie = (props) =>{
    let image = props.image ? <img className = "movie-img" src = {`${MOVIEDB_IMAGE_APPEND_URL}${props.image}`} alt = {props.movieTitle} />
                                     : <img className = "movie-img" src = {NO_IMAGE_URl} alt = {props.movieTitle} />;
    return (
        <div className = "col s12 m6 l3 movies-card">
            <div className = "card">
                <div className="card-image waves-effect waves-block waves-light">
                    {
                        image
                    }
                </div>
            </div>
            <div className="card-content">
                <button className="waves-effect waves-light btn-small overview" onClick = {() => props.viewInfoModal(props.movie) }>  Overview </button>
                <DeleteButton movie = {props.movie} onDeleteClick = {props.onDeleteClick}/>
            </div>
        </div>
    )
}

Movie.propTypes = { 
    image: PropTypes.string, 
    movieTitle: PropTypes.string,
    onDeleteClick: PropTypes.func,
    movie : PropTypes.object
} 

export default Movie;