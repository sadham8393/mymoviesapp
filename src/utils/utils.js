import { NOT_AVAILABLE } from './constants';

/**
 * Convert date format from "yyyy-mm-dd" to "dd/mm/yy"
 */

export function convertDate(date){
    if(date){
        return date.toString().split("-").reverse().join("/");
    }
    return NOT_AVAILABLE;
}