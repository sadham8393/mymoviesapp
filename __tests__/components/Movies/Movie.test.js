import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { findByTestClass, findByComponent } from '../../../utils/utils';
import Movie from '../../../src/components/Movies/Movie';
import DeleteButton  from "../../../src/components/Common/DeleteButton";

describe('<Movie /> component', () => { 
    let wrapper;
    let testProps;
    let mockFunc;
    beforeEach(() => {
        mockFunc = jest.fn();
        testProps = {
            image : "img",
            movieTitle : "Avengers",
            movie:{ title : "Avengers"},
            onDeleteClick : mockFunc,
            viewInfoModal : mockFunc
        };
        wrapper = shallow(<Movie {...testProps} />);
    });

    it('should render correctly', () => {
        const tree = renderer.create(<Movie {...testProps} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render movies-card correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.movies-card');
        expect(component.length).toBe(1);
    });

    it('should render movies-card correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.movies-card');
        expect(component.length).toBe(1);
    });

    it('should render card-image correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.card-image');
        expect(component.length).toBe(1);
    });

    it('should render movie-image correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.movie-img');
        expect(component.length).toBe(1);
    });

    it('should render card-content correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.card-content');
        expect(component.length).toBe(1);
    });

    it('should render overview button correctly and to be 1', () => {
        const component = findByTestClass(wrapper, '.overview');
        expect(component.length).toBe(1);
    });

    it('overview button Should emit callback on click event', () => {
        const button = findByTestClass(wrapper, '.overview');
        button.simulate('click');
        const callback = mockFunc.mock.calls.length;
        expect(callback).toBe(1);
    });

    it('should render deleteBtn correctly and to be 1', () => {
        const button = findByComponent(wrapper, DeleteButton);
        expect(button.length).toBe(1);
    });
});

