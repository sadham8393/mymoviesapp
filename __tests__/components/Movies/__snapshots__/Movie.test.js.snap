// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie /> component should render correctly 1`] = `
<div
  className="col s12 m6 l3 movies-card"
>
  <div
    className="card"
  >
    <div
      className="card-image waves-effect waves-block waves-light"
    >
      <img
        alt="Avengers"
        className="movie-img"
        src="https://image.tmdb.org/t/p/w500/img"
      />
    </div>
  </div>
  <div
    className="card-content"
  >
    <button
      className="waves-effect waves-light btn-small overview"
      onClick={[Function]}
    >
        Overview 
    </button>
    <button
      className="waves-effect waves-light btn-small deleteBtn"
      onClick={[Function]}
    >
       Delete 
    </button>
  </div>
</div>
`;
